pkg/consuldp/lifecycle_test.go

```diff
@@ -38,35 +40,35 @@ func TestLifecycleServerClosed(t *testing.T) {
 	require.Eventually(t, func() bool {
 		return !m.running
 	}, time.Second*2, time.Second)
-
 }
 
-func TestLifecycleServerEnabled(t *testing.T) {
+func TestLifecycleServer(t *testing.T) {
```

Comment:
**suggestion:** Even though there is some duplication in setup code, you may want to split this into two tests as you are doing startup specific things even when doing shutdown tests. I think it is making it more complicated than it needs to be.